xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -1532,27 +1532,49 @@ bool CGUIMediaWindow::OnPlayAndQueueMedia(const CFileItemPtr& item, const std::s
     // Remove ZIP, RAR files and folders
     CFileItemList playlist;
     playlist.Copy(*m_vecItems, true);
+
+    CLog::LogF(LOGDEBUG,
+               "m_vecItems (in playlist) before removal of folders/RAR/ZIP and duplicates");
+    for (const auto& p : playlist)
+      CLog::LogF(LOGDEBUG, "Path {}, DynPath {}", p->GetPath(), p->GetDynPath());
+
     playlist.erase(std::remove_if(playlist.begin(), playlist.end(),
                                   [](const std::shared_ptr<CFileItem>& i)
                                   { return i->IsZIP() || i->IsRAR() || i->m_bIsFolder; }),
                    playlist.end());
 
-    // Remove duplicates (eg. ISO/VIDEO_TS)
+    // Remove duplicates (eg. ISO/VIDEO_TS) in two stages
+    // Firstly with the same DynPath as our desired item
+    // Keeping our desired item instead of the first item with the same DynPath
+    playlist.erase(std::remove_if(playlist.begin(), playlist.end(),
+                                  [&item](const std::shared_ptr<CFileItem>& i) {
+                                    return i->GetDynPath() == item->GetDynPath() &&
+                                           i->GetPath() != item->GetPath();
+                                  }),
+                   playlist.end());
+
+    // Now remove all other duplicates
     playlist.erase(
         std::unique(playlist.begin(), playlist.end(),
-                    [](const std::shared_ptr<CFileItem>& i, const std::shared_ptr<CFileItem>& j) {
-                      return i->GetVideoInfoTag()->m_basePath == j->GetVideoInfoTag()->m_basePath;
-                    }),
+                    [](const std::shared_ptr<CFileItem>& i, const std::shared_ptr<CFileItem>& j)
+                    { return i->GetDynPath() == j->GetDynPath(); }),
         playlist.end());
 
+    CLog::LogF(LOGDEBUG, "Playlist after remove duplicates");
+    for (const auto& p : playlist)
+      CLog::LogF(LOGDEBUG, "Path {}, DynPath {}", p->GetPath(), p->GetDynPath());
+
     // Chosen item
     int mediaToPlay =
         std::distance(playlist.begin(), std::find_if(playlist.begin(), playlist.end(),
                                                      [&item](const std::shared_ptr<CFileItem>& i) {
-                                                       return i->GetVideoInfoTag()->m_basePath ==
-                                                              item->GetVideoInfoTag()->m_basePath;
+                                                       return i->GetDynPath() == item->GetDynPath();
```

Comment:
Why dynpath? Should by path. path == identity.

Reply:
I agree. Done.